packages/frontend/src/Components/HomePage/index.tsx

```diff
@@ -0,0 +1,176 @@
+         */
+
+        this.scrollObserver = {
+            callback: () => {
+                if (window.innerHeight + window.scrollY >= (
+                    window.document.body.scrollHeight - window.innerHeight) * 0.75
+                ) {
+                    this.updatePokemonsPage();
+                }
+            },
+        };
+    }
+
+    /** Insere this.scrollObserver no DOM */
+    private attachScrollObserver() {
+        if (!this.scrollObserverAttached) {
+            window.addEventListener("scroll", this.scrollObserver.callback);
+            this.scrollObserverAttached = true;
+        }
+    }
+
+
+    private async fetchPokemonsPage(
+        useCache = true,
+    ): Promise<{ data: Pokemon[], fromCache: boolean } | void> {
+        if (this.isFetching) {
+            return;
+        }
+
+        this.isFetching = true;
+        try {
+            const { paginationId } = this;
+
+            /** API com chave de cache */
+            const { api } = useCache ?
+                CachedApiFactory(`/Pokemon/getPage/${paginationId}`) :
+                this.apiFactory;
+
+
+            const response = await api.get(`/Pokemon/getPage/${paginationId}`);
+            this.isFetching = false;
+            // console.log(response);
+            const { fromCache } = response.request;
+
+            const { data } = response;
+
+            return { data, fromCache };
+
+        } catch (error) {
+            console.log(error);
+            toast.error(error.message);
+            this.isFetching = false;
+        }
+    }
+
+    private async updatePokemonsPage() {
+        if (this.isFetching) {
+            return;
+        }
+
+        this.fetchPokemonsPage().then(async (pokemons) => {
+            // Pega o ultimo elemento da Array pokemons (cuidado com off-by-one)
+            if (!pokemons) {
+                return;
+            }
+            const lastPokemon = pokemons.data[pokemons.data.length - 1];
+
+            let originalState: Array<Pokemon> | undefined = undefined;
+            if (pokemons.fromCache) {
+                // Vamos pegar do cache por agora, mas depois atualizaremos
+                this.isFetching = true;
+                originalState = this.state.pokemons;
+            } else {
+                // esse ítem já ta atualizado, avance para o próximo
+                this.paginationId = lastPokemon._id || '';
+            }
+
+            this.setState({ pokemons: [...(this.state.pokemons || []), ...pokemons.data] })
+
+            if (originalState) { // Se a requisição for retornada do cache
+                // Pegou do cache, faça a requisição novamente para atualizar
+                // os dados e evitar a "viagem no tempo" (conteúdo antigo)
+                // clearCacheByKey(`/Pokemon/getPage/${this.paginationId}`);
+                this.isFetching = false; // Se não a funçaão de fetch não roda
+                const { data } = await this.fetchPokemonsPage(false) || { data: [] };
```

Comment:
Não impacta em nada, mas o mais semânticamente correto seria
```suggestion
                const { data = []} = await this.fetchPokemonsPage(false);
```